internal/ingress/controller/store/store_test.go

```diff
@@ -605,35 +607,95 @@ func TestStore(t *testing.T) {
+
+	namespace := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "store-test",
+		},
+	}
+
+	ns, err := clientSet.CoreV1().Namespaces().Create(namespace)
 	if err != nil {
-		t.Errorf("unexpected error creating ingress client: %v", err)
+		t.Errorf("unexpected error creating the namespace: %v", err)
 	}
 	t.Logf("temporal namespace %v created", ns)
 
-	return ns
+	return ns.Name
 }
 
-func deleteNamespace(ns string, clientSet *kubernetes.Clientset, t *testing.T) {
-	t.Logf("deleting temporal namespace %v created", ns)
-	err := framework.DeleteKubeNamespace(clientSet, ns)
+func deleteNamespace(ns string, clientSet *fake.Clientset, t *testing.T) {
+	t.Logf("deleting temporal namespace %v", ns)
+
+	err := clientSet.CoreV1().Namespaces().Delete(ns, &metav1.DeleteOptions{})
 	if err != nil {
-		t.Errorf("unexpected error creating ingress client: %v", err)
+		t.Errorf("unexpected error deleting the namespace: %v", err)
 	}
 	t.Logf("temporal namespace %v deleted", ns)
 }
 
-func ensureIngress(ingress *extensions.Ingress, clientSet *kubernetes.Clientset) (*extensions.Ingress, error) {
-	s, err := clientSet.ExtensionsV1beta1().Ingresses(ingress.Namespace).Update(ingress)
+func createConfigMap(clientSet *fake.Clientset, ns string, t *testing.T) string {
+	t.Log("creating temporal config map")
+
+	configMap := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:     "config",
+			SelfLink: fmt.Sprintf("/api/v1/namespaces/%s/configmaps/config", ns),
+		},
+	}
+
+	cm, err := clientSet.CoreV1().ConfigMaps(ns).Create(configMap)
+	if err != nil {
+		t.Errorf("unexpected error creating the configuration map: %v", err)
+	}
+	t.Logf("required configmap %v created", cm)
+
+	return cm.Name
+}
+
+func deleteConfigMap(cm, ns string, clientSet *fake.Clientset, t *testing.T) {
+	t.Logf("deleting temporal configmap %v", cm)
+
+	err := clientSet.CoreV1().ConfigMaps(ns).Delete(cm, &metav1.DeleteOptions{})
+	if err != nil {
+		t.Errorf("unexpected error deleting the configmap: %v", err)
+	}
+	t.Logf("temporal configmap %v deleted", cm)
+}
+
+func ensureIngress(ingress *extensions.Ingress, clientSet *fake.Clientset, t *testing.T) *extensions.Ingress {
+	ing, err := clientSet.Extensions().Ingresses(ingress.Namespace).Update(ingress)
+
 	if err != nil {
 		if k8sErrors.IsNotFound(err) {
-			return clientSet.ExtensionsV1beta1().Ingresses(ingress.Namespace).Create(ingress)
+			t.Logf("ingress %v not found, creating", ingress)
+
+			ing, err = clientSet.Extensions().Ingresses(ingress.Namespace).Create(ingress)
+			if err != nil {
+				t.Fatalf("unexpected error creating ingress %+v: %v", ingress, err)
+			}
+
+			t.Logf("ingress %+v created", ingress)
+			return ing
 		}
-		return nil, err
+
+		t.Fatalf("unexpected error updating ingress %+v: %v", ingress, err)
 	}
-	return s, nil
+
+	t.Logf("ingress %+v updated", ingress)
+
+	return ing
+}
+
+func deleteIngress(ingress *extensions.Ingress, clientSet *fake.Clientset, t *testing.T) {
```

Comment:
This is a helper function, adding `t.Helper()` here would allow logs to report line numbers accurately.